### PR TITLE
[10.0][IMP] account_analytic_parent: propagate active through hierarchy

### PR DESCRIPTION
**Pull request by MiquelRForgeFlow**
Fix of https://github.com/OCA/account-analytic/pull/193.

@aheficent could you test it, please?

### CODE REVIEW

**Review by AaronHForgeFlow — APPROVED** [review]
:+1: Functional tests. Thanks.

**Review by LoisRForgeFlow — APPROVED** [review]
LGTM :+1:

**Comment by MiquelRForgeFlow** [comment]
@pedrobaeza

File: account_analytic_parent/models/account_analytic_account.py [hunk 1]
```diff
@@ -59,3 +59,25 @@ def name_get(self):
...
+
+    @api.multi
+    def archive_account(self):
+        for account in self:
+            account.active = False
+
+    @api.multi
+    def write(self, vals):
+        if 'active' in vals and not vals['active']:
+            self.mapped('child_ids').archive_account()
```
Comment (pedrobaeza) [inline account_analytic_parent/models/account_analytic_account.py]:
```suggestion
            self.mapped('child_ids').write({'active': False})
```
and remove extra method

**Review by pedrobaeza — CHANGES_REQUESTED** [review]
This doesn't propagate active to grandson and so

**Comment by MiquelRForgeFlow** [comment]
It propagates, because when you do `self.mapped('child_ids').write()`, then it will enter again in the `write` but for their children and thus will execute a `self.mapped('child_ids').write()` for the grandson and so on.

**Comment by pedrobaeza** [comment]
Right, thanks for the note.

**Comment by MiquelRForgeFlow** [comment]
ready :green_apple:

**Pull request merged**